### PR TITLE
Adherence records

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/models/ResourceList.java [hunk 1]
```diff
@@ -25,35 +25,45 @@
...
     public static final String MAX_REVISION = "maxRevision";
+    public static final String MIN_REVISION = "minRevision";
     public static final String NEXT_PAGE_OFFSET_KEY = "nextPageOffsetKey";
     public static final String NONE_OF_GROUPS = "noneOfGroups";
     public static final String OFFSET_BY = "offsetBy";
     public static final String OFFSET_KEY = "offsetKey";
     public static final String ORG_MEMBERSHIP = "orgMembership";
     public static final String PAGE_SIZE = "pageSize";
     public static final String PHONE_FILTER = "phoneFilter";
+    public static final String RECORD_TYPE = "recordType";
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/models/ResourceList.java]: record type -> adherence record type?

File: src/main/java/org/sagebionetworks/bridge/validators/AdherenceRecordsSearchValidator.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                errors.rejectValue(END_TIME_FIELD, END_TIME_MISSING);
+            } else {
+                if (search.getStartTime().isBefore(EARLIEST_DATETIME)) {
+                    errors.rejectValue(START_TIME_FIELD, BEFORE_EARLIEST_DATETIME_ERROR);
+                }
+                if (search.getEndTime().isAfter(LATEST_DATETIME)) {
+                    errors.rejectValue(END_TIME_FIELD, AFTER_LATEST_DATETIME_ERROR);
+                }
+                if (search.getEndTime().isBefore(search.getStartTime())) {
+                    errors.rejectValue(END_TIME_FIELD, END_TIME_BEFORE_START_TIME);
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/validators/AdherenceRecordsSearchValidator.java]: I mean, why Alx, let them search for a large time range if they want.

File: src/main/java/org/sagebionetworks/bridge/models/ResourceList.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/models/ResourceList.java]: Changed

File: src/main/java/org/sagebionetworks/bridge/validators/AdherenceRecordsSearchValidator.java (hunk 2 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/validators/AdherenceRecordsSearchValidator.java]: Removed

**Review by nategbrown9 — APPROVED** [review]

**Pull request merged**